### ISSUE TITLE
New line missing when "hitting" the window border

### ISSUE THREAD

**Issue by aranega** [billiob/terminology#33]
Hi there,

I don't know if I'm the only one, but since my last compilation, I have issues with long commands. Once the text reach the right border, it go left, but on the same line.

![issue](https://cloud.githubusercontent.com/assets/2317394/12961039/bd261354-d03e-11e5-87ed-2d4bd2576613.png)

If I continue typing, the next hit on the border drives the cursor at the good location. Once this happened, search through history are a little messy and shortcuts (like `ctlr+u`) have a tendency to introduce old chars in the input line.

**Comment by billiob** [comment]
Could you run a `git bisect` to find what commit introduced that behavior?

**Comment by aranega** [comment]
Sure, I'm on it.

**Comment by billiob** [comment]
(do not edit to add pieces of information, comment again or else I'm not notified of your changes)

It is sane. My efl/elementary are usually few weeks old. Maybe it's a bug introduced lately in efl/elementary.

**Comment by aranega** [comment]
(ok, my bad I thought edits were notified too)

I keep an eye on new EFL/Elementary commits thus, perhaps a fix will pop out :). I will report in this thread if anything change.

**Comment by aranega** [comment]
Ok, I finally figure it out. The issue is neither terminology (still work great) nor EFL/Elementary. It was "pew" (a virtualenvwrapper like for python virtual envs). I installed it the same day as I performed the terminology upgrade (in order to replace virtualenvwrapper) and "pew" adds a small code portion at the end of the `.basrc` in order to load code that print the current virtualenv (or nothing). This code is the reason the line is not break at the good place. 

For knowledge only, here is the code:

     PS1="\[\033[01;34m\]\$(basename '$VIRTUAL_ENV')\e[0m$PS1"

If I drop the line, not problems. The issue seems to come from the `\]` after the blue color selection. If I replace `\[\033[01;34m\]` by `\[\033[01;34m` everything is fine. But if I remove the first `\[`, the issue comes back. Is there any reason for that? Why the `\[` is required?
